crates/router/src/core/admin.rs

```diff
@@ -48,6 +48,15 @@ pub async fn create_merchant_account(
             })?,
     );
 
+    let primary_business_details = Some(
+        utils::Encode::<api::PrimaryBusinessDetails>::encode_to_value(
+            &req.primary_business_details,
+        )
+        .change_context(errors::ApiErrorResponse::InvalidDataValue {
+            field_name: "default business details",
+        })?,
+    );
```

Comment:
Wouldn't this also encode the `None` case as `Some(serde_json::Value::Null)` instead of directly being mapped as a `None`? Maybe use a `map()` instead?

Reply:
the entry in the database will be `null` in case its a `None` value